### ISSUE TITLE
m_k dimensionality

### ISSUE THREAD

**Issue by vanangamudi** [fuhuamosi/MatchLstm#3]
Hi Fuhuamosi,

in the MatchLstm model - [match_attention](https://github.com/fuhuamosi/MatchLstm/blob/master/model/match_lstm.py#L131) function, the shape of the tensor m_k -> [1, 2 * embedding_size] right? how the lsm_m operates over tensor of twice the size of its own state_size. Can you please help me understand?

        last_m_h = state.h
        sum_h = tf.matmul(h_s_j, w_s) + tf.matmul(h_t_k, w_t) + tf.matmul(last_m_h, w_m)
        e_kj = tf.matmul(tf.tanh(sum_h), w_e)
        a_kj = tf.nn.softmax(e_kj)
        alpha_k = tf.matmul(a_kj, h_s_j, transpose_a=True)
        alpha_k.set_shape([1, self._embedding_size])

        m_k = tf.concat([alpha_k, h_t_k], axis=1)
        with tf.variable_scope('{}_lstm_m'.format(self._name)):
            _, new_state = self.lstm_m(inputs=m_k, state=state)

**Comment by fuhuamosi** [comment]
You are right, the shape of the tensor m_k is  [1, 2 * embedding_size]. I believe that the hidden size of lstm smaller than embedding size is Okay.

**Comment by vanangamudi** [comment]
Thanks fuhuamosi. Maybe BasicLSTMCell transforms the input using a fully connected layer internally.
